TumbleBitSetup/PermutationTest/PermutationTestSetup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TumbleBitSetup
+{
+    public class PermutationTestSetup
+    {
+        public PermutationTestSetup()
+        {
+
+        }
+
+        public PermutationTestSetup(byte[] publicString, int alpha, int keySize, int k = 128)
```

Comment:
Yes, I thought that checking the size of the key should be done out of the scope of your proof.  But anyway, this does not hurt to check twice.